databroker/tests/test_v2/generic.py

```diff
@@ -28,26 +28,46 @@ def compare(a, b):
     a_indexed = {}
     b_indexed = {}
     for name, doc in a:
+        if name == 'resource':
+            # Check for an extraneous duplicate key in old documents.
+            if 'id' in doc:
+                assert doc['id'] == doc['uid']
+                doc = doc.copy()
+                doc.pop('id')
         if name == 'datum':
             a_indexed[('datum', doc['datum_id'])] = doc
+        # v0 yields {'_name": 'RunStop'} is the stop doc is missing; v2 yields None.
+        elif name == 'stop' and doc is None or 'uid' not in doc:
+            a_indexed[(name, None)] = None
         else:
             a_indexed[(name, doc['uid'])] = doc
     for name, doc in b:
+        if name == 'resource':
+            # Check for an extraneous duplicate key in old documents.
+            if 'id' in doc:
+                assert doc['id'] == doc['uid']
+                doc = doc.copy()
+                doc.pop('id')
         if name == 'datum':
```

Comment:
I think this should be an elif

Reply:
Ditto.

Reply:
See https://github.com/bluesky/databroker/pull/519#discussion_r380917557